### ISSUE TITLE
Upgrade Firefox 47.0 causes WebDriverException: Message: Connection reset

### ISSUE THREAD

**Issue by elgalu** [elgalu/docker-selenium#102]
After trying to upgrade Firefox from 46.0.1 to 47.0 I get this error on the automated tests:

```java
--------------------------
- Self test on firefox
--------------------------
video-rec: started
Will use browser=firefox
Will connect to selenium at http://127.0.0.1:24444/wd/hub
Will sleep '0.5' secs between test steps
Traceback (most recent call last):
  File "/usr/bin/python_test", line 40, in <module>
    driver = webdriver.Remote(command_executor=myselenium_hub_url, desired_capabilities=caps)
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/remote/webdriver.py", line 91, in __init__
    self.start_session(desired_capabilities, browser_profile)
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/remote/webdriver.py", line 178, in start_session
    response = self.execute(Command.NEW_SESSION, capabilities)
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/remote/webdriver.py", line 237, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/remote/errorhandler.py", line 194, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: Connection reset
Stacktrace:
    at java.net.SocketInputStream.read (SocketInputStream.java:209)
    at java.net.SocketInputStream.read (SocketInputStream.java:141)
    at org.apache.http.impl.io.SessionInputBufferImpl.streamRead (SessionInputBufferImpl.java:139)
    at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer (SessionInputBufferImpl.java:155)
    at org.apache.http.impl.io.SessionInputBufferImpl.readLine (SessionInputBufferImpl.java:284)
    at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead (DefaultHttpResponseParser.java:140)
    at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead (DefaultHttpResponseParser.java:57)
    at org.apache.http.impl.io.AbstractMessageParser.parse (AbstractMessageParser.java:261)
    at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader (DefaultBHttpClientConnection.java:165)
    at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader (CPoolProxy.java:167)
    at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse (HttpRequestExecutor.java:272)
    at org.apache.http.protocol.HttpRequestExecutor.execute (HttpRequestExecutor.java:124)
    at org.apache.http.impl.execchain.MainClientExec.execute (MainClientExec.java:271)
    at org.apache.http.impl.execchain.ProtocolExec.execute (ProtocolExec.java:184)
    at org.apache.http.impl.execchain.RetryExec.execute (RetryExec.java:88)
    at org.apache.http.impl.execchain.RedirectExec.execute (RedirectExec.java:110)
    at org.apache.http.impl.client.InternalHttpClient.doExecute (InternalHttpClient.java:184)
    at org.apache.http.impl.client.CloseableHttpClient.execute (CloseableHttpClient.java:71)
    at org.apache.http.impl.client.CloseableHttpClient.execute (CloseableHttpClient.java:55)
    at org.openqa.selenium.remote.internal.ApacheHttpClient.fallBackExecute (ApacheHttpClient.java:144)
    at org.openqa.selenium.remote.internal.ApacheHttpClient.execute (ApacheHttpClient.java:90)
    at org.openqa.selenium.remote.HttpCommandExecutor.execute (HttpCommandExecutor.java:142)
    at org.openqa.selenium.firefox.internal.NewProfileExtensionConnection.execute (NewProfileExtensionConnection.java:160)
    at org.openqa.selenium.firefox.FirefoxDriver$LazyCommandExecutor.execute (FirefoxDriver.java:380)
    at org.openqa.selenium.remote.RemoteWebDriver.execute (RemoteWebDriver.java:644)
    at org.openqa.selenium.remote.RemoteWebDriver.startSession (RemoteWebDriver.java:249)
    at org.openqa.selenium.remote.RemoteWebDriver.<init> (RemoteWebDriver.java:131)
    at org.openqa.selenium.firefox.FirefoxDriver.<init> (FirefoxDriver.java:218)
    at org.openqa.selenium.firefox.FirefoxDriver.<init> (FirefoxDriver.java:211)
    at org.openqa.selenium.firefox.FirefoxDriver.<init> (FirefoxDriver.java:129)
    at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0 (NativeConstructorAccessorImpl.java:-2)
    at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance (Constructor.java:453)
    at org.openqa.selenium.remote.server.FirefoxDriverProvider.callConstructor (FirefoxDriverProvider.java:91)
    at org.openqa.selenium.remote.server.FirefoxDriverProvider.newInstance (FirefoxDriverProvider.java:68)
    at org.openqa.selenium.remote.server.DefaultDriverFactory.newInstance (DefaultDriverFactory.java:60)
    at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call (DefaultSession.java:222)
    at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call (DefaultSession.java:1)
    at java.util.concurrent.FutureTask.run (FutureTask.java:266)
    at org.openqa.selenium.remote.server.DefaultSession$1.run (DefaultSession.java:176)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1158)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:632)
    at java.lang.Thread.run (Thread.java:804)
```

**Comment by elgalu** [comment]
This looks like a regression https://github.com/seleniumhq/selenium-google-code-issue-archive/issues/7819

**Comment by elgalu** [comment]
Ping @lukeis

**Comment by lukeis** [comment]
see seleniumhq/selenium#2110

**Comment by elgalu** [comment]
Fixed with Firefox 47.0.1 plus Selenium 2.53.1 in release https://github.com/elgalu/docker-selenium/releases/tag/2.53.1a
